<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Craig Orman, Naga Vempati, Heike Hofmann">
<meta name="dcterms.date" content="2022-12-09">

<title>Comparative Analysis: features of full and cropped scans</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="Comparative Analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="Comparative Analysis_files/libs/quarto-html/quarto.js"></script>
<script src="Comparative Analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="Comparative Analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Comparative Analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="Comparative Analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Comparative Analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Comparative Analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Comparative Analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Comparative Analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Comparative Analysis: features of full and cropped scans</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Craig Orman, Naga Vempati, Heike Hofmann </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 9, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<p>Naga, if you have any problems running this file, exchange line 6 with</p>
<p>bibliography: “zotero.bib”</p>
<p>If that doesn’t work either, holler on slack</p>
<div class="cell" data-hash="Comparative-Analysis_cache/html/data_7d66ce80e136fed47532623d6ef55e53">

</div>
<section id="comparison-of-features-on-full-and-cropped-scans" class="level1">
<h1>Comparison of features on Full and Cropped Scans</h1>
<section id="abstract" class="level3">
<h3 class="anchored" data-anchor-id="abstract">Abstract</h3>
<p>Deciding if a scan’s quality is often difficult, and time consuming if a scan needs to be redone. The scanner, often not familiar with the statistical processes to follow, needs an evaluater to consider each scan’s quality before it leaves the microscope mount. We propose using the X3P scan quality indicator, and in this comparative analysis, we evaluate four of the features used to determine if a scan is of sufficient quality. We found that two of the features give superior results when calculated on a cropped version of an image, one of them is impartial to cropped or full image, and the last feature is best ran over the full image.</p>
<p>To conduct our analysis, we did a series of tests to compare the cropped version of a scan against the full version. We constructed scatter and box plots as well as ROC curves for visual analysis We trained a Generalized Linear Model (glm) to test the p scores, or importance, of each variable in the model, and constructed kernel density graphs to compare the scans. Every cropped scan and its full scan counterpart were found to have more than 90% correlation with each other, requiring us to use only one of them for each feature to avoid co-linearity issues.</p>
<p><a href="#extract-na">Extract NA</a>, a measure of the total missing value count in the scan over the total value count. This feature performed better when it is cropped with no reservations. The ROC curve and p score in the glm were both better for the cropped image. The kernel density graph was also more distinct for the cropped image.</p>
<p><a href="#assess-bottomempty">Assess Bottomempty</a>, a measure of missing value count in the bottom 20% of the scan compared to the total value count in the same area. This feature performed better when it is cropped with no reservations. The ROC curve and p score in the glm were both better for the cropped image. The kernel density graph was also more distinct for the cropped image.</p>
<p><a href="#assess-col-na">Assess Col NA</a>, the proportion of columns in the scans matrix which have more than 20% missing values. This feature performed only marginally better when cropped. The ROC area difference is only 0.008, and the p scores for both features were found to be extremely significant. The kernal density graphs are very similar with the cropped graph being slightly more distinct.</p>
<p><a href="#assess-median-na-proportion">Assess Median NA Proportion</a>, calculates the mean number of NA’s in each column, and then finds the median out of all those values. This feature performed better when left as the full scan. The ROC area for the cropped was better by only 0.001, the p values for both features were found to be extremely significant, but the full scan was lower. The kernel density graph was more distinct for the full images.</p>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-2_0e34945f34e5da51048a55019aa55c21">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 31%">
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 17%">
<col style="width: 10%">
<col style="width: 13%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Correlation</th>
<th style="text-align: left;">pvalue_Full</th>
<th style="text-align: left;">pvalue_cropped</th>
<th style="text-align: left;">auc_Full</th>
<th style="text-align: left;">auc_Cropped</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Extract NA</td>
<td style="text-align: left;">0.908</td>
<td style="text-align: left;">0.229</td>
<td style="text-align: left;">&lt;2e-16</td>
<td style="text-align: left;">0.871</td>
<td style="text-align: left;">0.902</td>
</tr>
<tr class="even">
<td style="text-align: left;">Assess Bottomempty</td>
<td style="text-align: left;">0.905</td>
<td style="text-align: left;">5.86e-10</td>
<td style="text-align: left;">&lt;2e-16</td>
<td style="text-align: left;">0.783</td>
<td style="text-align: left;">0.859</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Assess Col NA</td>
<td style="text-align: left;">0.919</td>
<td style="text-align: left;">1.62e-06</td>
<td style="text-align: left;">1.17e-14</td>
<td style="text-align: left;">0.888</td>
<td style="text-align: left;">0.896</td>
</tr>
<tr class="even">
<td style="text-align: left;">Assess Median NA Proportion</td>
<td style="text-align: left;">0.908</td>
<td style="text-align: left;">&lt;2e-16</td>
<td style="text-align: left;">6.77e-10</td>
<td style="text-align: left;">0.907</td>
<td style="text-align: left;">0.863</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>All of the features had significant outliers, which we have called “<a href="#followup-scans">Followup Scans</a>”, for their individual predictive power. We investigated the type I errors, false positives of a bad scan being predicted as good, and found them all to be “Tiny Problems” scans which could reasonably be re-classified as “problematic” or worse.</p>
<p>XXX Related Works - <span style="color: red;">Do we need forensics specific papers here, or can we use general imagery comparative analysis?</span></p>
<p><span style="color: darkorange;">Good question, and I don’t know - I’m not sure that there is any kind of forensic specific work out (yet). I’m waiting for a NIST paper to finally come out :) </span></p>
</section>
<section id="background-and-introduction" class="level2">
<h2 class="anchored" data-anchor-id="background-and-introduction">Background and Introduction</h2>
<p>XXX General stylistic comments</p>
<ul>
<li><span style="color: darkorange;">there are no ’you’s in a scientific paper. Make sure to be specific of each actor.</span></li>
</ul>
<p>Relevant citations for criticism of the subjective nature of current firearms comparisons: <span class="citation" data-cites="presidentscouncilofadvisorsonscienceandtechnologyForensicScienceCriminal2016 nationalresearchcouncilu.s.StrengtheningForensicScience2009">(<a href="#ref-presidentscouncilofadvisorsonscienceandtechnologyForensicScienceCriminal2016" role="doc-biblioref">President’s Council of Advisors on Science and Technology 2016</a>; <a href="#ref-nationalresearchcouncilu.s.StrengtheningForensicScience2009" role="doc-biblioref">National Research Council (U.S.) 2009</a>)</span></p>
<p><span class="citation" data-cites="presidentscouncilofadvisorsonscienceandtechnologyForensicScienceCriminal2016">President’s Council of Advisors on Science and Technology (<a href="#ref-presidentscouncilofadvisorsonscienceandtechnologyForensicScienceCriminal2016" role="doc-biblioref">2016</a>)</span>,<br>
<span class="citation" data-cites="nationalresearchcouncilu.s.StrengtheningForensicScience2009">National Research Council (U.S.) (<a href="#ref-nationalresearchcouncilu.s.StrengtheningForensicScience2009" role="doc-biblioref">2009</a>)</span></p>
<p>XXX Writing in progress - <span style="color: red;">Some questions on this section, am I being too generic? Do I need to add more sources in this section?</span></p>
<p><span style="color: darkorange;">Generally, it looks good, but we can probably get away with fewer words and more images :)</span></p>
<p>In many court cases, determining the source of evidence is crucial, whether it is finding the correct murder weapon, or proving which cartridge casing came from which firearm. The particular evidence of interest to this study is determining if a bullet is fired from a particular firearm.</p>
<p>Currently, the method used to determine if a specific bullet came from a specific firearm requires that a bullet is found or recovered during the investigation, preferably as undamaged as possible, and the firearm. An examiner would fire the weapon into a kevlar tube so that they had a second bullet to use as the control, or a known match to the barrel. Most modern firearms have rifling, which is a series of spiral grooves in a barrel that cause the bullet to spin, leading to greater speed, accuracy, and range. Rifling is done via mass production, but due to very small inconsistencies, and the use and treatment of the barrel after production, the rifling leaves striation marks on the bullets. Here, we are only interested in striation marks on land-engraved areas (LEAs). Striation marks show strong similarities between bullets from different firings through the same barrel, to the degree that they are considered in the forensic community to be ‘unique’ to the particular barrel <span class="citation" data-cites="pageUniquenessForensicIdentification2011">(<a href="#ref-pageUniquenessForensicIdentification2011" role="doc-biblioref">Page, Taylor, and Blenkin 2011</a>)</span>, but a general assessment of the random matching probability has so far proven to be elusive. The grooves on the bullets are called striations and the general pattern is similar across all bullets fired from a particular barrel. Two main flaws in this method are that barrels are usually interchangeable on firearms, and that continued firing from the barrel will alter the chance of a match.</p>
<p>Forensic Examiners are the current standard for comparing if two bullets have sufficient striation matching to be considered a positive match. A forensic examiner is a person who uses a comparison microscope to view two of the bullets and their striations and matches them. The problem with using forensic examiners is that they are still subjective[Report to the president], and have been proven to have atleast higher than a 1% error rate. <span class="citation" data-cites="presidentscouncilofadvisorsonscienceandtechnologyForensicScienceCriminal2016 nationalresearchcouncilu.s.StrengtheningForensicScience2009">(<a href="#ref-presidentscouncilofadvisorsonscienceandtechnologyForensicScienceCriminal2016" role="doc-biblioref">President’s Council of Advisors on Science and Technology 2016</a>; <a href="#ref-nationalresearchcouncilu.s.StrengtheningForensicScience2009" role="doc-biblioref">National Research Council (U.S.) 2009</a>)</span></p>
<p>To further aid in forensic abilities and quantitative methods, CSAFE has been researching ways to use machine learning and other methods to automatically and quantitatively compare bullet striation marks. <span class="citation" data-cites="vanderplasComparisonThreeSimilarity2020">(<a href="#ref-vanderplasComparisonThreeSimilarity2020" role="doc-biblioref">Vanderplas et al. 2020</a>)</span>, which compares bullets using a variety of statistical and mathematical models to generate confidence scores of the similarities of striation markings. Automated methods and forensic examiners alike have a similar issue that this paper will hereby address. The problem is differing quality of microscopy scans. In order to compare the scans, each bullet must be under a microscope and a digital scan be made. Scanning of the bullets can go wrong in a variety of ways, from lighting conditions, bullet placement in the holders, to actual damage to the bullets surface. In order to detect when there are flaws in the scanning process, we have created an RShiny app that reads in the scanned X3P files, and gives a confidence score, with 100 being a perfect, un-flawed scan, to 0 being a terrible scan.</p>
<p>To increase the speed and performance of our quality scanner, we decided to crop the X3P files, this helps eliminate the ‘noise’ around the images that can skew the accuracy of the results. By using smaller images, we also increase the speed of evaluation, as there is less to process.</p>
<p>This analysis is to compare the difference between the cropped versus non-cropped (full) version of a scan for quality identification. Cropped images have the potential for decreasing noise around the signal. The level of cropping we are considering is 5% from the left and right sides, and 10% off of the bottom of the image. In particular, we want to preserve the bottom of the image and the center as that is where most of the signal is. Below are examples of a full, full with marked edges, and a cropped image.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU-254-BB-L1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Image of a scan without cropping</figcaption><p></p>
</figure>
</div>
<ul>
<li><span style="color: red;">Is this the kind of image you were imagining? I think we need to make the red lines beefier and more obvious</span></li>
</ul>
<p><img src="./Comparative-Analysis_files/figure-html/FAU-254-BB-L1-masked.png" class="img-fluid" alt="Image of a scan with the edges colored"> <img src="./Comparative-Analysis_files/figure-html/FAU-254-BB-L1-Cropped.png" class="img-fluid" alt="Image of a scan with cropping"></p>
</section>
<section id="features" class="level2">
<h2 class="anchored" data-anchor-id="features">Features</h2>
<p>Each of the features shares a similar set up, so the following assumptions and definitions will remain independently true for each of the features defined below.</p>
<p>Let <span class="math inline">\(A=\{NA\}\)</span> be the set of undefined values. For simplicity of notation we will assume that the space of real values <span class="math inline">\({\rm I\!R}\)</span> contains <span class="math inline">\(A\)</span>: <span class="math inline">\({\rm I\!R}:= {\rm I\!R} \cup A\)</span>.</p>
<p>Let <span class="math inline">\(X \in R^{m,n}\)</span> be a real-valued surface matrix of dimensions m x n where m and n are strictly positive integers <span class="math inline">\(X = (x_{ij})_{1 \leq i \leq m, 1 \leq j \leq n}\)</span>.</p>
<p><a href="#extract-na">Extract NA</a></p>
<p>The function <code>extract_na</code> calculates the percentage of missing values in the scan (part) under observation, e.g.&nbsp;for scan surface matrix <span class="math inline">\(X \in {\rm I\!R}^{m, n}\)</span> the percentage of missing values is defined as:</p>
<p>The proportion of missing values in X is then defined as: <span class="math display">\[
\frac{1}{m*n} \sum^m_{i=1} \sum^n_{j=1} \theta_A(x_{ij}) \\
\text{Where } \theta_A(x) = \left\{\begin{aligned}
&amp;1 &amp;&amp;: \text{if }x \in A\\
&amp;0 &amp;&amp;: \text{otherwise}\\
\end{aligned} \right.
\]</span> <a href="#assess-bottomempty">Assess Bottomempty</a></p>
<p>The feature <code>assess_bottomempty</code> calculates the percentage of missing values in the bottom 20% of the scan.</p>
<p>Let <span class="math inline">\(R \subseteq {\rm I\!R}\)</span> be a set of size m, where each element is the sum of the NA’s for the given row, defined as: <span class="math display">\[
\forall i \in R: R_i = \sum^n_{j=1} \theta_A(x_{ij}) \\
\text{Where } \theta_A(x) = \left\{\begin{aligned}
&amp;1 &amp;&amp;: \text{if }x \in A\\
&amp;0 &amp;&amp;: \text{otherwise}\\
\end{aligned} \right.
\]</span></p>
<p>Let <span class="math inline">\(B \subset R\)</span> be a set, which is the set of all values in <span class="math inline">\(R_i\)</span>, given that <span class="math inline">\(i \geq m*0.8\)</span>. Therefore, the proportion of missing values in <span class="math inline">\(X\)</span>’s bottom 20% can be given by: <span class="math display">\[
\frac{1}{m*n*0.2}\sum_{i=1}^{m*0.2}(R_i)*100
\]</span> <a href="#assess-col-na">Assess Col NA</a></p>
<p>The function <code>assess_col_na</code> is the proportion of columns in the image matrix which have more NA’s than 20%.</p>
<p>For every column in the matrix of a scan, we find the proportion of scans in that column which are NA. Then we count how many of the columns whose proportion is greater than 20%, the pre-determined threshold of acceptable NA’s. Then we divide by the number of columns * 0.2 to get our final threshold adjusted number.</p>
<p>Let <span class="math inline">\(R \subseteq {\rm I\!R}\)</span> be a set of size n, where each element is the sum of the NA’s for the given column, defined as: <span class="math display">\[
\forall i \in R: R_i = \sum^m_{j=1} \theta_A(x_{ij}) \\
\text{Where } \theta_A(x) = \left\{\begin{aligned}
&amp;1 &amp;&amp;: \text{if }x \in A\\
&amp;0 &amp;&amp;: \text{otherwise}\\
\end{aligned} \right.
\]</span></p>
<p>We define <span class="math inline">\(P\)</span> as the proportion of all NAs per column for every row, as defined here: <span class="math display">\[
\forall i \in R: P_i = \frac{R_i}{n} * 100
\]</span></p>
<p>We now find the proportion of threshold adjusted columns in the matrix <span class="math display">\[
\frac{\sum_{i=1}^n(P_i*\beta_B(P_i))}{n*0.2} \\
\text{Where } \beta_B(x) = \left\{\begin{aligned}
&amp;1 &amp;&amp;: \text{if }x &gt; 20\\
&amp;0 &amp;&amp;: \text{otherwise}\\
\end{aligned} \right.
\]</span></p>
<p><a href="#assess-median-na-proportion">Assess Median NA proportion</a></p>
<p>The function <code>assess_median_na_proportion</code> calculates the mean number of NA’s in each column, and then finds the median out of all those values.</p>
<p>Let <span class="math inline">\(R \subseteq {\rm I\!R}\)</span> be a set of size n, where each element is the mean of the NA’s for the given column, defined as: <span class="math display">\[
\forall i \in R: R_i = \frac{\sum^m_{j=1} \theta_A(x_{ij})}{m} \\
\text{Where } \theta_A(x) = \left\{\begin{aligned}
&amp;1 &amp;&amp;: \text{if }x \in A\\
&amp;0 &amp;&amp;: \text{otherwise}\\
\end{aligned} \right.
\]</span></p>
<p>We then sort, and select the median of <span class="math inline">\(R\)</span></p>
</section>
<section id="appendix" class="level2">
<h2 class="anchored" data-anchor-id="appendix">Appendix</h2>
<section id="extract-na" class="level3">
<h3 class="anchored" data-anchor-id="extract-na">Extract NA</h3>
<section id="which-of-the-features-is-better-for-discriminating-between-good-and-bad-scans" class="level4">
<h4 class="anchored" data-anchor-id="which-of-the-features-is-better-for-discriminating-between-good-and-bad-scans">Which of the features is better for discriminating between good and bad scans?</h4>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-3_c12bccb8bdba72c5f51f2e27d73da990">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Returning more (or less) than 1 row per `summarise()` group was deprecated in
dplyr 1.1.0.
ℹ Please use `reframe()` instead.
ℹ When switching from `summarise()` to `reframe()`, remember that `reframe()`
  always returns an ungrouped data frame and adjust accordingly.
ℹ The deprecated feature was likely used in the yardstick package.
  Please report the issue at &lt;https://github.com/tidymodels/yardstick/issues&gt;.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-3-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Extract NA. Correlation:  0.908 Full AUC: 0.871 Cropped AUC:  0.902"</code></pre>
</div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Summary Info of Extract NA</caption>
<colgroup>
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 13%">
<col style="width: 13%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">min</th>
<th style="text-align: right;">firstQ</th>
<th style="text-align: right;">med</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">thirdQ</th>
<th style="text-align: right;">max</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Standard</td>
<td style="text-align: right;">3.666645</td>
<td style="text-align: right;">12.79727</td>
<td style="text-align: right;">15.082281</td>
<td style="text-align: right;">15.915471</td>
<td style="text-align: right;">18.30296</td>
<td style="text-align: right;">48.45372</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cropped</td>
<td style="text-align: right;">1.389464</td>
<td style="text-align: right;">6.38338</td>
<td style="text-align: right;">8.060005</td>
<td style="text-align: right;">8.916349</td>
<td style="text-align: right;">10.57698</td>
<td style="text-align: right;">40.80471</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="should-we-use-features-from-just-one-type-of-scan-or-both" class="level4">
<h4 class="anchored" data-anchor-id="should-we-use-features-from-just-one-type-of-scan-or-both">Should we use features from just one type of scan or both?</h4>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-4_4284726bf67f25e0263a305be04f3e9b">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = GoodScan ~ extract_na + extract_na_cropped, family = binomial(), 
    data = full_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-3.1882  -0.2394   0.3044   0.5298   4.0795  

Coefficients:
                   Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)         7.16247    0.37858  18.919   &lt;2e-16 ***
extract_na         -0.04349    0.03613  -1.204    0.229    
extract_na_cropped -0.58303    0.05020 -11.615   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 2230.6  on 1849  degrees of freedom
Residual deviance: 1299.2  on 1847  degrees of freedom
AIC: 1305.2

Number of Fisher Scoring iterations: 6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="conclusion-for-extract-na" class="level4">
<h4 class="anchored" data-anchor-id="conclusion-for-extract-na">Conclusion for Extract NA</h4>
<p>The values for feature <code>extract_NA</code> are highly correlated between the cropped and the full scan.</p>
<p>Using good and scans with only tiny problems as overall ‘good’ scans, the feature applied to cropped scans has an increased accuracy compared to the feature values from the full scan.</p>
<p>We might want to follow up on the orange colored scans:</p>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-5_e17709bc94f18fa9ea0a1b8f0d47c14d">
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-6_e8924f196a82ed4bd09442ee0a20b927">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>full_data<span class="sc">$</span>LAPD_id[full_data<span class="sc">$</span>followup]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FAU263-BA-L4" "FAU263-BC-L1" "FAU263-BC-L3" "FAU287-BC-L5" "FAU154-BD-L2"
[6] "FAU277-BA-L4" "FAU286-BA-L5"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>followupScans <span class="ot">&lt;-</span> <span class="fu">rbind</span>(followupScans, full_data[full_data<span class="sc">$</span>followup <span class="sc">==</span> <span class="cn">TRUE</span>,])</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># All followups for extract_na are mislabelled scans. They are all labelled as tiny problems but should be problematic or worse.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="assess-bottomempty" class="level3">
<h3 class="anchored" data-anchor-id="assess-bottomempty">Assess Bottomempty</h3>
<section id="which-of-the-features-is-better-for-discriminating-between-good-and-bad-scans-1" class="level4">
<h4 class="anchored" data-anchor-id="which-of-the-features-is-better-for-discriminating-between-good-and-bad-scans-1">Which of the features is better for discriminating between good and bad scans?</h4>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-8_95ea0728698a7cb8f884ffa01e1b8a78">
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-8-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-8-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Assess Bottomempty. Correlation:  0.905 Full AUC: 0.783 Cropped AUC:  0.859"</code></pre>
</div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Summary Info of Assess Bottomempty</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">min</th>
<th style="text-align: right;">firstQ</th>
<th style="text-align: right;">med</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">thirdQ</th>
<th style="text-align: right;">max</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Standard</td>
<td style="text-align: right;">8.516493</td>
<td style="text-align: right;">22.66947</td>
<td style="text-align: right;">27.54019</td>
<td style="text-align: right;">29.90259</td>
<td style="text-align: right;">34.63373</td>
<td style="text-align: right;">95.39375</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cropped</td>
<td style="text-align: right;">3.564558</td>
<td style="text-align: right;">10.78581</td>
<td style="text-align: right;">13.54205</td>
<td style="text-align: right;">15.28335</td>
<td style="text-align: right;">17.71191</td>
<td style="text-align: right;">84.22476</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="should-we-use-features-from-just-one-type-of-scan-or-both-1" class="level4">
<h4 class="anchored" data-anchor-id="should-we-use-features-from-just-one-type-of-scan-or-both-1">Should we use features from just one type of scan or both?</h4>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-9_d53d3fafbc1e39f4403c8c1750bdd78d">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = GoodScan ~ assess_bottomempty + assess_bottomempty_cropped, 
    family = binomial(), data = full_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.6794  -0.2813   0.4084   0.6028   4.0680  

Coefficients:
                           Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)                 4.57647    0.25156  18.193  &lt; 2e-16 ***
assess_bottomempty          0.09046    0.01460   6.194 5.86e-10 ***
assess_bottomempty_cropped -0.40779    0.02738 -14.895  &lt; 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 2230.6  on 1849  degrees of freedom
Residual deviance: 1503.7  on 1847  degrees of freedom
AIC: 1509.7

Number of Fisher Scoring iterations: 5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="conclusion-for-assess-bottomempty" class="level4">
<h4 class="anchored" data-anchor-id="conclusion-for-assess-bottomempty">Conclusion for Assess Bottomempty</h4>
<p>The values for feature <code>assess_bottomempty</code> are highly correlated between the cropped and the full scan.</p>
<p>Using good and scans with only tiny problems as overall ‘good’ scans, the feature applied to cropped scans has an increased accuracy compared to the feature values from the full scan.</p>
<p>We might want to follow up on the orange colored scans:</p>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-10_8345dbde5eb1ab9117083ed3068b412c">
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-11_2084c19b43aba876e870c3175bbe7334">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>full_data<span class="sc">$</span>LAPD_id[full_data<span class="sc">$</span>followup]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FAU263-BA-L4" "FAU287-BC-L5" "FAU254-BD-L4" "FAU275-BC-L5" "FAU275-BD-L3"
[6] "FAU277-BA-L4" "FAU286-BA-L5"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>followupScans <span class="ot">&lt;-</span> <span class="fu">rbind</span>(followupScans, full_data[full_data<span class="sc">$</span>followup <span class="sc">==</span> <span class="cn">TRUE</span>,])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="assess-col-na" class="level3">
<h3 class="anchored" data-anchor-id="assess-col-na">Assess Col NA</h3>
<section id="which-of-the-features-is-better-for-discriminating-between-good-and-bad-scans-2" class="level4">
<h4 class="anchored" data-anchor-id="which-of-the-features-is-better-for-discriminating-between-good-and-bad-scans-2">Which of the features is better for discriminating between good and bad scans?</h4>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-13_da26c06d0c67a3b531301872c527bd16">
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-13-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Assess Col NA Correlation:  0.919 Full AUC: 0.888 Cropped AUC:  0.896"</code></pre>
</div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Summary Info of Assess Col NA</caption>
<colgroup>
<col style="width: 13%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 13%">
<col style="width: 13%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">min</th>
<th style="text-align: right;">firstQ</th>
<th style="text-align: right;">med</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">thirdQ</th>
<th style="text-align: right;">max</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Standard</td>
<td style="text-align: right;">0.2573318</td>
<td style="text-align: right;">0.9244176</td>
<td style="text-align: right;">1.0755102</td>
<td style="text-align: right;">1.1654392</td>
<td style="text-align: right;">1.292844</td>
<td style="text-align: right;">4.368436</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cropped</td>
<td style="text-align: right;">0.0920783</td>
<td style="text-align: right;">0.4700281</td>
<td style="text-align: right;">0.6126759</td>
<td style="text-align: right;">0.6851693</td>
<td style="text-align: right;">0.807577</td>
<td style="text-align: right;">3.356120</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="should-we-use-features-from-just-one-type-of-scan-or-both-2" class="level4">
<h4 class="anchored" data-anchor-id="should-we-use-features-from-just-one-type-of-scan-or-both-2">Should we use features from just one type of scan or both?</h4>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-14_3e8cda166736c677a6064181d4563654">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = GoodScan ~ assess_col_na + assess_col_na_cropped, 
    family = binomial(), data = full_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-3.2254  -0.1897   0.3176   0.5365   3.6049  

Coefficients:
                      Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)             7.3581     0.3785  19.440  &lt; 2e-16 ***
assess_col_na          -2.4724     0.5155  -4.796 1.62e-06 ***
assess_col_na_cropped  -4.8233     0.6249  -7.719 1.17e-14 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 2230.6  on 1849  degrees of freedom
Residual deviance: 1308.8  on 1847  degrees of freedom
AIC: 1314.8

Number of Fisher Scoring iterations: 6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="conclusion-for-assess-col-na" class="level4">
<h4 class="anchored" data-anchor-id="conclusion-for-assess-col-na">Conclusion for Assess Col NA</h4>
<p>The values for feature <code>assess_col_na</code> are highly correlated between the cropped and the full scan.</p>
<p>Using good and scans with only tiny problems as overall ‘good’ scans, the feature applied to cropped scans has an increased accuracy compared to the feature values from the full scan.</p>
<p>We might want to follow up on the orange colored scans:</p>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-15_b55cbcf10c4be1b0e6546f59359b46c1">
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-16_19a8d77af84a6f1ff86668d26d94479a">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>full_data<span class="sc">$</span>LAPD_id[full_data<span class="sc">$</span>followup]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FAU263-BA-L4" "FAU263-BB-L3" "FAU263-BC-L1" "FAU263-BC-L3" "FAU154-BD-L2"
[6] "FAU286-BA-L5"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>followupScans <span class="ot">&lt;-</span> <span class="fu">rbind</span>(followupScans, full_data[full_data<span class="sc">$</span>followup <span class="sc">==</span> <span class="cn">TRUE</span>,])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="assess-median-na-proportion" class="level3">
<h3 class="anchored" data-anchor-id="assess-median-na-proportion">Assess Median NA Proportion</h3>
<section id="which-of-the-features-is-better-for-discriminating-between-good-and-bad-scans-3" class="level4">
<h4 class="anchored" data-anchor-id="which-of-the-features-is-better-for-discriminating-between-good-and-bad-scans-3">Which of the features is better for discriminating between good and bad scans?</h4>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-18_e23cde23dec63848b541e5141234f170">
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-18-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-18-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Assess Median NA Proportion. Correlation:  0.908 Full AUC: 0.907 Cropped AUC:  0.863"</code></pre>
</div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Summary Info of Assess median NA proportion</caption>
<colgroup>
<col style="width: 13%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">min</th>
<th style="text-align: right;">firstQ</th>
<th style="text-align: right;">med</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">thirdQ</th>
<th style="text-align: right;">max</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Standard</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0033099</td>
<td style="text-align: right;">0.0108771</td>
<td style="text-align: right;">0.0207348</td>
<td style="text-align: right;">0.0271577</td>
<td style="text-align: right;">0.2552491</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cropped</td>
<td style="text-align: right;">0.0011692</td>
<td style="text-align: right;">0.0518201</td>
<td style="text-align: right;">0.0686747</td>
<td style="text-align: right;">0.0751096</td>
<td style="text-align: right;">0.0918769</td>
<td style="text-align: right;">0.2994505</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="should-we-use-features-from-just-one-type-of-scan-or-both-3" class="level4">
<h4 class="anchored" data-anchor-id="should-we-use-features-from-just-one-type-of-scan-or-both-3">Should we use features from just one type of scan or both?</h4>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-19_277ef18d2b5abb9e4059949c9978420f">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = GoodScan ~ assess_median_na_proportion + assess_median_na_proportion_cropped, 
    family = binomial(), data = full_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.8226  -0.1079   0.3143   0.4865   5.2612  

Coefficients:
                                    Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)                            4.384      0.257  17.060  &lt; 2e-16 ***
assess_median_na_proportion          -83.035      5.925 -14.014  &lt; 2e-16 ***
assess_median_na_proportion_cropped  -21.546      3.491  -6.171 6.77e-10 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 2230.6  on 1849  degrees of freedom
Residual deviance: 1219.0  on 1847  degrees of freedom
AIC: 1225

Number of Fisher Scoring iterations: 6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="conclusion-for-assess-median-na-proportion" class="level4">
<h4 class="anchored" data-anchor-id="conclusion-for-assess-median-na-proportion">Conclusion for Assess Median NA Proportion</h4>
<p>The values for feature <code>extract_NA</code> are highly correlated between the cropped and the full scan.</p>
<p>Using good and scans with only tiny problems as overall ‘good’ scans, the feature applied to full scans has an increased accuracy compared to the feature values from the cropped scan.</p>
<p>We might want to follow up on the orange colored scans:</p>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-20_8a028bbb37c9bd45b0ffc1c2189c5dad">
<div class="cell-output-display">
<p><img src="Comparative-Analysis_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-21_ac249d886db35abf42760cc62722f122">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>full_data<span class="sc">$</span>LAPD_id[full_data<span class="sc">$</span>followup]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "FAU263-BC-L3" "FAU154-BD-L2" "FAU204-BC-L4"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>followupScans <span class="ot">&lt;-</span> <span class="fu">rbind</span>(followupScans, full_data[full_data<span class="sc">$</span>followup <span class="sc">==</span> <span class="cn">TRUE</span>,])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="followup-scans" class="level3">
<h3 class="anchored" data-anchor-id="followup-scans">Followup Scans</h3>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-23_16de15ed84b4072e07bc862cdd73bf45">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>followupUnique <span class="ot">&lt;-</span> followupScans[<span class="fu">duplicated</span>(followupScans) <span class="sc">==</span> <span class="cn">FALSE</span>,]</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>followupScans <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(followupScans<span class="sc">$</span>LAPD_id) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">count =</span> <span class="fu">n</span>()</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 12 × 2
   `followupScans$LAPD_id` count
   &lt;chr&gt;                   &lt;int&gt;
 1 FAU154-BD-L2                3
 2 FAU204-BC-L4                1
 3 FAU254-BD-L4                1
 4 FAU263-BA-L4                3
 5 FAU263-BB-L3                1
 6 FAU263-BC-L1                2
 7 FAU263-BC-L3                3
 8 FAU275-BC-L5                1
 9 FAU275-BD-L3                1
10 FAU277-BA-L4                2
11 FAU286-BA-L5                3
12 FAU287-BC-L5                2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 3 hits: FAU154-BD-L2, FAU263-BA-L4, FAU263-BC-L3, FAU286-BA-L5</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 hits: FAU263-BC-L1, FAU277-BA-L4, FAU287-BC-L5</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 1 hits: FAU204-BC-L4, FAU254-BD-L4, FAU263-BB-L3, FAU275-BC-L5, FAU275-BD-L3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-24_d20e7e5c9195acd6a222739a7aeead93">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>FAU263_BA_L4 <span class="ot">&lt;-</span> <span class="fu">x3p_read</span>(<span class="st">"../data/followup_scans/LAPD - 263 - Bullet A - Land 4 - Sneox2 - 20x - auto light left image + 20% - threshold 2 - resolution 4 - Connor Hergenreter.x3p"</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>FAU263_BC_L1 <span class="ot">&lt;-</span> <span class="fu">x3p_read</span>(<span class="st">"../data/followup_scans/LAPD - 263 - Bullet C - Land 1 - Sneox2 - 20x - auto light left image + 20% - threshold 2 - resolution 4 - Connor Hergenreter.x3p"</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>FAU263_BC_L3 <span class="ot">&lt;-</span> <span class="fu">x3p_read</span>(<span class="st">"../data/followup_scans/LAPD - 263 - Bullet C - Land 3 - Sneox2 - 20x - auto light left image + 20% - threshold 2 - resolution 4 - Connor Hergenreter.x3p"</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>FAU277_BA_L4 <span class="ot">&lt;-</span> <span class="fu">x3p_read</span>(<span class="st">"../data/followup_scans/LAPD - 277 - Bullet A - Land 4 - Sneox2 - 20x - auto light left image + 20% - threshold 2 - resolution 4 - Connor Hergenreter.x3p"</span>)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">x3p_image</span>(FAU263_BA_L4, <span class="at">file=</span><span class="st">"./Comparative-Analysis_files/figure-html/FAU263_BA_L4.png"</span>)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="fu">x3p_image</span>(FAU263_BC_L1, <span class="at">file=</span><span class="st">"./Comparative-Analysis_files/figure-html/FAU263-BC-L1.png"</span>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="fu">x3p_image</span>(FAU263_BC_L3, <span class="at">file=</span><span class="st">"./Comparative-Analysis_files/figure-html/FAU263_BC_L3.png"</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="fu">x3p_image</span>(FAU277_BA_L4, <span class="at">file=</span><span class="st">"./Comparative-Analysis_files/figure-html/FAU277-BA-L4.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-hash="Comparative-Analysis_cache/html/unnamed-chunk-25_9edafa0957e7ab40931bedd9c59e7a3a">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 17%">
<col style="width: 13%">
<col style="width: 21%">
<col style="width: 21%">
<col style="width: 26%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">LAPD.ID</th>
<th style="text-align: right;">Hit.Count</th>
<th style="text-align: left;">Current.Quality</th>
<th style="text-align: left;">Current.Problem</th>
<th style="text-align: left;">Recommended.Quality</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">FAU263-BA-L4</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Feathering</td>
<td style="text-align: left;">Bad</td>
</tr>
<tr class="even">
<td style="text-align: left;">FAU263-BC-L3</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Feathering</td>
<td style="text-align: left;">Bad</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FAU154-BD-L2</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Holes</td>
<td style="text-align: left;">Bad</td>
</tr>
<tr class="even">
<td style="text-align: left;">FAU286-BA-L5</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Holes</td>
<td style="text-align: left;">Bad</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FAU263-BC-L1</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Feathering</td>
<td style="text-align: left;">Bad</td>
</tr>
<tr class="even">
<td style="text-align: left;">FAU287-BC-L5</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Feathering</td>
<td style="text-align: left;">Problematic</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FAU277-BA-L4</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Holes</td>
<td style="text-align: left;">Problematic</td>
</tr>
<tr class="even">
<td style="text-align: left;">FAU254-BD-L4</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Holes</td>
<td style="text-align: left;">Problematic</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FAU275-BC-L5</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Holes</td>
<td style="text-align: left;">Problematic</td>
</tr>
<tr class="even">
<td style="text-align: left;">FAU275-BD-L3</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Holes</td>
<td style="text-align: left;">Problematic</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FAU263-BB-L3</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Feathering</td>
<td style="text-align: left;">Problematic</td>
</tr>
<tr class="even">
<td style="text-align: left;">FAU204-BC-L4</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Tiny Problems</td>
<td style="text-align: left;">Holes</td>
<td style="text-align: left;">Bad</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU263_BA_L4.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU263_BA_L4 Image</figcaption><p></p>
</figure>
</div>
<p>FAU263_BA_L4 (3 hits, Tiny Problems, Feathering): Significant feathering on right side image, missing most of the left, and many missing values on bottom.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU263_BC_L3.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU263_BC_L3 Image</figcaption><p></p>
</figure>
</div>
<p>FAU263_BC_L3 (3 hits, Tiny Problems, Feathering): Contains significant feathering, holes, disproportionate edges and missing values at the bottom.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU154_BD_L2.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU154_BD_L2 Image</figcaption><p></p>
</figure>
</div>
<p>FAU154_BD_L2 (3 hits, Tiny Problems, Holes): Significant feathering across image.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU286_BA_L5.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU286_BA_L5 Image</figcaption><p></p>
</figure>
</div>
<p>FAU286_BA_L5 (3 hits, Tiny Problems, Holes): Significant missing values on the left, right, bottom right and center. Holes throughout center.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU263-BC-L1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU263_BC_L1 Image</figcaption><p></p>
</figure>
</div>
<p>FAU263_BC_L1 (2 hits, Tiny Problems, Feathering): Significant feathering on right hand side, left side is missing most of the values, then feathering, then holes as it moves towards the middle. Bottom is also speckled with missing values.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU287_BC_L5.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU287_BC_L5 Image</figcaption><p></p>
</figure>
</div>
<p>FAU287_BC-L5 (2 hits, Tiny Problems, Feathering): Significant hole on right side, feathering on left and right sides, minor holes in the center.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU277_BA_L4.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU277_BA_L4 Image</figcaption><p></p>
</figure>
</div>
<p>FAU277_BA_L4 (2 hits, Tiny Problems, Holes): A few holes, significant missing values on the left, right, and bottom.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU254_BD_L4.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU254_BD_L4 Image</figcaption><p></p>
</figure>
</div>
<p>FAU254_BD_L4 (1 hits, Tiny Problems, Holes): Significant missing values on the bottom, holes across the center, missing section on right side.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU275_BC_L5.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU275_BC_L5 Image</figcaption><p></p>
</figure>
</div>
<p>FAU275_BC_L5 (1 hits, Tiny Problems, Holes): Missing a large section on the right and bottom. Holes in middle.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU275_BD_L3.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU275_BD_L3 Image</figcaption><p></p>
</figure>
</div>
<p>FAU275_BD_L3 (1 hits, Tiny Problems, Holes): Significant number of missing values on the left and right, as well as the bottom. Holes in center.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU263_BB_L3.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU263_BB_L3 Image</figcaption><p></p>
</figure>
</div>
<p>FAU263_BB_L3 (1 hits, Tiny Problems, Holes): Significant missing values on bottom, left, and right. Feathering on the right and a few holes in the middle.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./Comparative-Analysis_files/figure-html/FAU204_BC_L4.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">FAU204_BC_L4 Image</figcaption><p></p>
</figure>
</div>
<p>FAU204_BC_L4 (1 hits, Tiny Problems, Holes): Feathering on each end of image, rotational issues on left edge, holes in the center</p>

</section>
</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-nationalresearchcouncilu.s.StrengtheningForensicScience2009" class="csl-entry" role="doc-biblioentry">
National Research Council (U.S.), ed. 2009. <em>Strengthening Forensic Science in the <span>United States</span>: A Path Forward</em>. <span>Washington, D.C</span>: <span>National Academies Press</span>.
</div>
<div id="ref-pageUniquenessForensicIdentification2011" class="csl-entry" role="doc-biblioentry">
Page, Mark, Jane Taylor, and Matt Blenkin. 2011. <span>“Uniqueness in the Forensic Identification Sciences–Fact or Fiction?”</span> <em>Forensic Science International</em> 206 (1-3): 12–18. <a href="https://doi.org/10.1016/j.forsciint.2010.08.004">https://doi.org/10.1016/j.forsciint.2010.08.004</a>.
</div>
<div id="ref-presidentscouncilofadvisorsonscienceandtechnologyForensicScienceCriminal2016" class="csl-entry" role="doc-biblioentry">
President’s Council of Advisors on Science and Technology. 2016. <span>“Forensic <span>Science</span> in <span>Criminal Courts</span>: <span>Ensuring Scientific Validity</span> of <span>Feature Comparison Methods</span>.”</span> <a href="https://obamawhitehouse.archives.gov/sites/default/files/microsites/ostp/PCAST/pcast_forensic_science_report_final.pdf">https://obamawhitehouse.archives.gov/sites/default/files/microsites/ostp/PCAST/pcast_forensic_science_report_final.pdf</a>.
</div>
<div id="ref-vanderplasComparisonThreeSimilarity2020" class="csl-entry" role="doc-biblioentry">
Vanderplas, Susan, Melissa Nally, Tylor Klep, Cristina Cadevall, and Heike Hofmann. 2020. <span>“Comparison of Three Similarity Scores for Bullet <span>LEA</span> Matching.”</span> <em>Forensic Science International</em>, January, 110167. <a href="https://doi.org/gn6487">https://doi.org/gn6487</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>